rmw_iceoryx_cpp/src/rmw_compare_guids_equal.cpp

```diff
@@ -27,10 +29,22 @@ rmw_compare_gids_equal(
 {
   *result = false;
 
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(gid1, RMW_RET_ERROR);
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(gid2, RMW_RET_ERROR);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(gid1, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    gid1,
+    gid1->implementation_identifier,
+    rmw_get_implementation_identifier(),
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(gid2, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    gid2,
+    gid2->implementation_identifier,
+    rmw_get_implementation_identifier(),
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
 
-  RMW_SET_ERROR_MSG("rmw_iceoryx_cpp does not support gids.");
-  return RMW_RET_UNSUPPORTED;
+  if (std::memcmp(gid1->data, gid2->data, sizeof(gid1->data)) == 0) {
```

Comment:
While this also works, it seems at other places `RMW_GID_STORAGE_SIZE` was used instead of `sizeof(gid1->data)`